lib/src/notification/notification.component.ts

```diff
@@ -0,0 +1,177 @@
+import { OnChanges, Component, ViewEncapsulation, Input, Output, EventEmitter, OnInit, SimpleChanges, OnDestroy } from "@angular/core";
+
+export type NotificationStyle = "slide-in" | "bar";
+export type NotificationPosition = "bottom-left" | "bottom-right" | "top-left" | "top-right" | "top" | "bottom";
+export type NotificationTheme = "purple" | "primary" | "danger" | "success" | "warning" | "inverted";
+
+export interface NotificationAction {
+    text: string;
+    action: () => void;
+}
+
+@Component({
+    selector: "sebng-notification",
+    templateUrl: "./notification.component.html",
+    styleUrls: ["./notification.component.scss"],
+    encapsulation: ViewEncapsulation.None,
+})
+export class NotificationComponent implements OnChanges, OnInit, OnDestroy {
+    @Input() className?: string;
+    @Input() dismissable?: boolean;
+    @Input() dismissTimeout?: number;
+    @Input() message?: string;
+    @Output() onClick?: EventEmitter<MouseEvent> = new EventEmitter();
+
+    @Output() onDismiss?: EventEmitter<void> = new EventEmitter();
+
+    @Input() persist?: boolean;
+    @Input() position?: NotificationPosition;
+    @Input() style?: NotificationStyle;
+    @Input() theme?: NotificationTheme;
+    @Input() title?: string;
+    @Input() toggle: boolean;
+    @Input() actions?: Array<NotificationAction>;
+
+    public notificationClassNames: string;
+    public showNotificationBody: boolean = false;
+    public showNotificationTitle: boolean = false;
+    private timerRef: number = null;
+    private defaultTimeout: number = 5000;
+
+    // helper functions
+
+    /**
+     * Get the style class based on the theme passed through the props
+     * @param {string} style The style passed through the props
+     * @returns {void} The style class
+     */
+    private getStyleClass(): void {
+        let styleClass: string = "style-";
+        if (this.style && ["slide-in", "bar"].indexOf(this.style) !== -1) {
+            styleClass += this.style;
+        } else {
+            styleClass += "slide-in";
+        }
+
+        this.notificationClassNames += " " + styleClass;
+
+        this.showNotificationTitle = this.title && this.style === ("style-slide-in" as NotificationStyle);
```

Comment:
wouldn't this always returns `false` since `this.style` will not have `style-` prefix defined by the user?

Maybe this is what it should look like:
```suggestion
        this.showNotificationTitle = this.title && styleClass === "style-slide-in";
```

Reply:
Yah this is crucial, the unit test ought have failed too.